kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/metrics/MessageMetrics.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.internal.metrics;
+
+import java.util.Optional;
+
+import org.apache.kafka.common.protocol.ApiKeys;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.Meter.MeterProvider;
+import io.netty.channel.ChannelDuplexHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelPromise;
+
+import io.kroxylicious.proxy.frame.Frame;
+import io.kroxylicious.proxy.internal.util.Metrics;
+
+public class MessageMetrics extends ChannelDuplexHandler {
```

Comment:
I should add a couple of unit tests for these.  I guess I can use the Netty Embedded Channel.

Reply:
Done.